File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        
        whites = 0
        for i in range(k):
            if blocks[i] == "W":
                whites += 1

        ans = whites
        n= len(blocks)
        for i in range(k,n):
            if blocks[i] == "W":
                whites += 1
            if blocks[i-k] == "W":
                whites -= 1
            ans = min(ans,whites)

        return ans

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {

        int whites = 0;
        int ans = 1e7;
        int n = blocks.size();
        for (int i = 0; i < n; ++i) {
            if (blocks[i] == 'W') ++whites;
            if (i >= k and blocks[i-k] == 'W') --whites;
            if (i >= k-1) ans = min(ans,whites);
        }

        return ans;
        
    }
};